l10n_ec_einvoice/einvoice.py

```diff
@@ -0,0 +1,338 @@
+                totalConImpuestos.append(totalImpuesto)
+
+        infoFactura.append(totalConImpuestos)
+
+        etree.SubElement(infoFactura, 'propina').text = '0.00'
+        etree.SubElement(infoFactura, 'importeTotal').text = '{:.2f}'.format(invoice.amount_pay)
+        etree.SubElement(infoFactura, 'moneda').text = 'DOLAR'
+
+        if invoice.state == 'paid':
+            pagos = self.get_payment_element(cr, uid, invoice)
+            infoFactura.append(pagos)        
+
+        return infoFactura
+
+    def _get_detail_element(self, cr, uid, invoice):
+        """
+        """
+        cur_obj = self.pool.get('res.currency')        
+        detalles = etree.Element('detalles')
+        company_currency = self.pool['res.company'].browse(cr, uid, invoice.company_id.id).currency_id        
+        for line in invoice.invoice_line:
+            if line.product_id.default_code == 'DESC':
+                continue
+            detalle = etree.Element('detalle')
+            etree.SubElement(detalle, 'codigoPrincipal').text = line.product_id.default_code
+            if line.product_id.manufacturer_pref:
+                etree.SubElement(detalle, 'codigoAuxiliar').text = line.product_id.manufacturer_pref
+            etree.SubElement(detalle, 'descripcion').text = line.product_id.name
+            etree.SubElement(detalle, 'cantidad').text = '%.6f' % (line.quantity)
+            etree.SubElement(detalle, 'precioUnitario').text = '%.6f' % (line.price_unit)
+            etree.SubElement(detalle, 'descuento').text = '0.0000'
+            etree.SubElement(detalle, 'precioTotalSinImpuesto').text = '%.2f' % (line.price_subtotal)
+            impuestos = etree.Element('impuestos')
+            for tax_line in line.invoice_line_tax_id:
+                if tax_line.tax_group in ['vat', 'vat0', 'ice', 'other']:
+                    base_amount = cur_obj.compute(cr, uid, invoice.currency_id.id, company_currency, line.price_subtotal * tax_line.base_sign, context={'date': invoice.date_invoice or time.strftime('%Y-%m-%d')}, round=False)
+                    tax_amount = cur_obj.compute(cr, uid, invoice.currency_id.id, company_currency, tax_line.amount * tax_line.tax_sign, context={'date': invoice.date_invoice or time.strftime('%Y-%m-%d')}, round=False)
+                    impuesto = etree.Element('impuesto')
+                    etree.SubElement(impuesto, 'codigo').text = codigoImpuesto[tax_line.tax_group]
+                    etree.SubElement(impuesto, 'codigoPorcentaje').text = tarifaImpuesto[tax_line.tax_group]
+                    etree.SubElement(impuesto, 'tarifa').text = '%.2f' % (tax_amount * 100)
+                    etree.SubElement(impuesto, 'baseImponible').text = '%.2f' % (base_amount)
+                    etree.SubElement(impuesto, 'valor').text = '%.2f' % (base_amount * tax_amount)
+                    impuestos.append(impuesto)
+            detalle.append(impuestos)
```

Comment:
@miltonlab es para evitar el calculo en este for y solo leer los valores de los campos.
@alcidesrivera es importante detalle que campos se requieren exactamente (con retencion, con iva, etc) para implementarlos correctamente.